backend/app/operations/lesson_operations.py

```diff
@@ -89,39 +89,50 @@ def get_lessons_by_class(class_: Class) -> list[Lesson]:
 
 
 @with_instance(User)
-def get_student_lessons(user: User, sort: bool = True) -> list[Lesson]:
+def get_student_lessons(user: User, limit : int = None, sort: bool = True) -> list[Lesson]:
     """Get all lessons associated with a student through class groups."""
     # Collect lessons from all groups where the student is enrolled
     lessons = []
     for group in user.groups:
-        lessons.extend(group.lessons)
-
-    return sorted(lessons) if sort else lessons
+        all_lessons = group.lessons 
+        lessons.extend(all_lessons)
+    
+    if limit is not None:
+        return sorted(lessons[:limit]) if sort else lessons[:limit]
+    else :
+        return sorted(lessons) if sort else lessons
 
-@with_instance(User)
-def get_student_future_lessons(user: User, sort: bool = True) -> list[Lesson]:
+@with_instance(User)  
+def get_student_future_lessons(user: User, limit: int = None, sort: bool = True) -> list[Lesson]:
     """Get future lessons for a student through class groups."""
     # Retrieve all lessons for the student from their groups
     future_lessons = []
     date = datetime.now().date()
     for group in user.groups:
-        group_lessons = [lesson for lesson in group.lessons if lesson.date >= date]
+        all_lesson=group.lessons
+        group_lessons = [lesson for lesson in all_lesson if lesson.date >= date]
+
+    if limit is not None :
```

Comment:
Don't think this should be here
`group_lessons` was defined in loop. Just use the limit in the final return statement